### DISCUSSION TITLE
Boiler energy values in 3.6.3 - how are they determined?

### DISCUSSION THREAD

**Discussion by gwilford** [emsesp/EMS-ESP32#1441] [Q&A] [answered]
Hi, I'm liking the new DHW and heating energy values provided by ems-esp. I had been estimating them myself using instantaneous boiler power (modulation) and nominal power and the delta values provided by ems-esp are similar while the absolute values are not.

However, the ems-esp energy sensors in 3.6.3 are reading 10-20% higher than my actual smart gas meter readings. How does ems-esp determine the values? Are they taken direct from the boiler or calculated using a formula?

**Comment by MichaelDvP** [comment, marked as answer]
See #1148 and #1268

**Reply by gwilford** [reply]
Thanks. In summary, calculated using nominal power and modulation values...
